hist/histpainter/src/THistPainter.cxx

```diff
@@ -10549,51 +10538,49 @@ Int_t THistPainter::TableInit()
 const char * THistPainter::GetBestFormat(Double_t v, Double_t e, const char *f)
 {
 
-   static char ef[20];
-   char tf[20], tv[64];
+   TString ef, tf, tv;
 
    // print v with the format f in tv.
-   snprintf(tf,20,"%s%s","%",f);
-   snprintf(tv,64,tf,v);
+   tf = TString::Format("%s%s","%",f);
+   tv = TString::Format(tf.Data(),v);
 
    // Analyse tv.
-   TString sv = tv;
-   int ie = sv.Index("e");
-   int iE = sv.Index("E");
-   int id = sv.Index(".");
+   int ie = tv.Index("e");
+   int iE = tv.Index("E");
+   int id = tv.Index(".");
 
    // v has been printed with the exponent notation.
    // There is 2 cases, the exponent is positive or negative
    if (ie >= 0 || iE >= 0) {
-      if (sv.Index("+") >= 0) {
+      if (tv.Index("+") >= 0) {
          if (e < 1) {
-            snprintf(ef,20,"%s.1f","%");
+            ef = TString::Format("%s.1f","%");
          } else {
             if (ie >= 0) {
-               snprintf(ef,20,"%s.%de","%",ie-id-1);
+               ef = TString::Format("%s.%de","%",ie-id-1);
             } else {
-               snprintf(ef,20,"%s.%dE","%",iE-id-1);
+               ef = TString::Format("%s.%dE","%",iE-id-1);
             }
          }
       } else {
          if (ie >= 0) {
-            snprintf(ef,20,"%s.%de","%",ie-id-1);
+            ef = TString::Format("%s.%de","%",ie-id-1);
          } else {
-            snprintf(ef,20,"%s.%dE","%",iE-id-1);
+            ef = TString::Format("%s.%dE","%",iE-id-1);
          }
       }
 
    // There is not '.' in tv. e will be printed with one decimal digit.
    } else if (id < 0) {
-      snprintf(ef,20,"%s.1f","%");
+      ef = TString::Format("%s.1f","%");
 
    // There is a '.' in tv and no exponent notation. e's decimal part will
    // have the same number of digits as v's one.
    } else {
-      snprintf(ef,20,"%s.%df","%",sv.Length()-id-1);
+      ef = TString::Format("%s.%df","%",tv.Length()-id-1);
    }
 
-   return ef;
+   return Form("%s",ef.Data());
```

Comment:
Here same problem as with `GetObjectInfo` - return global variable which may be destroyed before it really used. While here is static function, just use `static TString ef;` and `return ef.Data();`